quarkus-solace-messaging-connector/runtime/src/main/java/com/solace/quarkus/messaging/outgoing/SolaceDirectMessageOutgoingChannel.java

```diff
@@ -0,0 +1,285 @@
+                        metadata.getPartitionKey());
+            }
+
+            if (metadata.getDynamicDestination() != null) {
+                topic.set(Topic.of(metadata.getDynamicDestination()));
+            }
+        });
+
+        Object payload = m.getPayload();
+        if (payload instanceof OutboundMessage) {
+            outboundMessage = (OutboundMessage) payload;
+        } else if (payload instanceof String) {
+            outboundMessage = msgBuilder
+                    .withHTTPContentHeader(HttpHeaderValues.TEXT_PLAIN.toString(), "")
+                    .build((String) payload);
+        } else if (payload instanceof byte[]) {
+            outboundMessage = msgBuilder.build((byte[]) payload);
+        } else {
+            outboundMessage = msgBuilder
+                    .withHTTPContentHeader(HttpHeaderValues.APPLICATION_JSON.toString(), "")
+                    .build(Json.encode(payload));
+        }
+
+        if (isTracingEnabled) {
+            SolaceTrace solaceTrace = new SolaceTrace.Builder()
+                    .withDestinationKind("topic")
+                    .withTopic(topic.get().getName())
+                    .withMessageID(outboundMessage.getApplicationMessageId())
+                    .withCorrelationID(outboundMessage.getCorrelationId())
+                    .withPartitionKey(
+                            outboundMessage
+                                    .hasProperty(SolaceConstants.MessageUserPropertyConstants.QUEUE_PARTITION_KEY)
+                                            ? outboundMessage
+                                                    .getProperty(
+                                                            SolaceConstants.MessageUserPropertyConstants.QUEUE_PARTITION_KEY)
+                                            : null)
+                    .withPayloadSize(Long.valueOf(outboundMessage.getPayloadAsBytes().length))
+                    .withProperties(outboundMessage.getProperties()).build();
+            solaceOpenTelemetryInstrumenter.traceOutgoing(m, solaceTrace);
+        }
+
+        return Uni.createFrom().<Object> emitter(e -> {
+            boolean exitExceptionally = false;
+            try {
+                if (isPublisherReady) {
+                    publisher.publish(outboundMessage, topic.get());
+                    if (waitForPublishReceipt) {
+                        publishStatus = new CompletableFuture<>();
+                        publisher.setPublishFailureListener(failedPublishEvent -> {
+                            publishStatus = CompletableFuture.completedFuture(true);
+                            PubSubPlusClientException exception = failedPublishEvent.getException();
+                            e.fail(exception);
+                        });
+                        try {
+                            publishStatus.get(publishReceiptTimeout, TimeUnit.MILLISECONDS);
```

Comment:
@ozangunalp Direct messaging will not receive any acknowledgement from broker as it is fire and forget. However we can check for any publish failures. So to simulate waitForPublishReceipt I have enabled a configuration to wait for any failure else emit success

Reply:
I am not sure if `waitForPublishReceipt` makes sense for direct publishing. 
Setting the failure listener on each message send may fail the wrong message. 
Maybe we should set the failure listener once and just log the message failure and always ack direct outgoing messages ?
Otherwise, another option could be to generate a correlation id property for each direct outgoing message and use it to correlate publish failure and set the correct message sending on failure?

Reply:
Yeah I thought of using correlation id but the challenge is we need to have access to emitter object and the correlation-id outside the scope of this method. With that we will be able to fail or complete the emitter.

Otherwise better option would be complete the emitter by default but set the health status based on failure event? We can mention this in documentation in direct messaging section.

Reply:
To use a correlation id you'd need to generate and attach it to the outgoing message, and access it on the `FailedPublishEvent`. 

> Otherwise better option would be complete the emitter by default but set the health status based on failure event? We can mention this in documentation in direct messaging section.

I think so

Reply:
Yes, but still the emitter object is not accessible. Probably will implement second option for now